### PR TITLE
add more docs

### PR DESCRIPTION
**Pull request by quasiben**

### CODE REVIEW

**Comment by quasiben** [comment]
@jakirkham would you be willing to look this over ?

File: docs/source/quickstart.rst [hunk 1]
```diff
@@ -0,0 +1,108 @@
+Quickstart
```
Comment (quasiben) [inline docs/source/quickstart.rst]: @madsbk what do you think about this example for a quickstart ?

File: docs/source/configuration.rst [hunk 2]
```diff
@@ -1,44 +1,74 @@
...
 Env Vars
-----------
+--------
 
 DEBUG
 ~~~~~
 
+Debug variables for both UCX and UCX-PY can be set
 
 ``UCX_PY_LOG_LEVEL``
```
Comment (madsbk) [inline docs/source/configuration.rst]: `UCX_PY_LOG_LEVEL` => `UCXPY_LOG_LEVEL`

File: docs/source/quickstart.rst [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+::
+
+    git clone https://github.com/openucx/ucx
+    cd ucx
+    ./autogen.sh
+    mkdir build
+    cd build
+    ../configure --prefix=$CONDA_PREFIX --enable-debug --with-cuda=$CUDA_HOME --enable-mt CPPFLAGS="-I//$CUDA_HOME/include"
+    make -j install
```
Comment (madsbk) [inline docs/source/quickstart.rst]: I would change this to:
```bash
git clone https://github.com/openucx/ucx
cd ucx
./autogen.sh
mkdir build
cd build

# Performance build
../contrib/configure-release --prefix=$CONDA_PREFIX --with-cuda=$CUDA_HOME --enable-mt CPPFLAGS="-I/$CUDA_HOME/include"
# Debug build
../contrib/configure-devel --prefix=$CONDA_PREFIX --with-cuda=$CUDA_HOME --enable-mt CPPFLAGS="-I/$CUDA_HOME/include"

make -j install

File: docs/source/quickstart.rst [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+2) Install UCX-PY
+
+::
+
+    git clone git@github.com:rapidsai/ucx-py.git
+    cd ucx-py
+    46python setup.py build_ext --inplace
+    python -m pip install -e .
```
Comment (madsbk) [inline docs/source/quickstart.rst]: I would change this to:
```python
git clone https://github.com/rapidsai/ucx-py.git
cd ucx-py
pip install .
# or for develop build
pip install -v -e .
```

File: docs/source/quickstart.rst (hunk 1 above)
Comment (madsbk) [reply docs/source/quickstart.rst]: Looks good; I have added some suggestions.

File: docs/source/quickstart.rst [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+::
+
+    conda create -n ucx -c conda-forge -c jakirkham/label/ucx \
+    cudatoolkit=<CUDA version> ucx-proc=*=gpu ucx ucx-py python=3.7
+
+Source
+~~~~~~
+
+The following instructions assume you'll be using ucx-py on a CUDA enabled system.
```
Comment (madsbk) [inline docs/source/quickstart.rst]: Add `... and is in a Conda environment`

**Comment by quasiben** [comment]
Thanks @madsbk

File: docs/source/configuration.rst [hunk 6]
```diff
@@ -1,44 +1,72 @@
 Configuration
 =============
 
+UCX can be configured with a wide variety of options and optimizations including: transport, caching, etc.
+
+For novice user we recommend the following settings:
+
+::
+
+    UCX_MEMTYPE_CACHE=n UCX_TLS=all
```
Comment (jakirkham) [inline docs/source/configuration.rst]: Should we add an explanation of `UCX_MEMTYPE_CACHE` below as well?

File: docs/source/configuration.rst [hunk 7]
```diff
@@ -1,44 +1,72 @@
...
 
-Values:
-- rc = ibv_post_send, ibv_post_recv, ibv_poll_cq
-- cuda_copy = cuMemHostRegister, cuMemcpyAsync
-- cuda_ipc =  cuIpcCloseMemHandle ,  cuIpcOpenMemHandle, cuMemcpyAsync
-- sockcm = connection management over sockets
-- tcp = communication over TCP
+Transport Methods (Simplified):
+
+- ``rc`` -> (IB) ibv_post_send, ibv_post_recv, ibv_poll_cq
```
Comment (jakirkham) [inline docs/source/configuration.rst]: Should we just say using InfiniBand perhaps with these functions in parentheses?

File: docs/source/configuration.rst [hunk 8]
```diff
@@ -1,44 +1,72 @@
...
+- ``cuda_copy`` -> cuMemHostRegister, cuMemcpyAsync
```
Comment (jakirkham) [inline docs/source/configuration.rst]: Should we mention Unified Virtual Addressing here?

File: docs/source/configuration.rst [hunk 9]
```diff
@@ -1,44 +1,72 @@
...
+- ``cuda_ipc`` -> (NVLINK) cuIpcCloseMemHandle, cuIpcOpenMemHandle, cuMemcpyAsync
```
Comment (jakirkham) [inline docs/source/configuration.rst]: Might be worth stating CUDA Interprocess Communication explicitly and then relegating the functions to a parenthesized list afterwards.

File: docs/source/configuration.rst [hunk 10]
```diff
@@ -1,44 +1,72 @@
...
+- ``sockcm`` -> connection management over sockets
+- ``tcp`` -> TCP over sockets (often used with sockcm)
+
+
+InfiniBand Device
+~~~~~~~~~~~~~~~~~~
+
+Select InfiniBand Device
+
+``UCX_NET_DEVICES``
```
Comment (jakirkham) [inline docs/source/configuration.rst]: Should we show a short example here?

File: docs/source/quickstart.rst [hunk 11]
```diff
@@ -0,0 +1,114 @@
+Quickstart
+==========
+
+
+Install
+-------
```
Comment (jakirkham) [inline docs/source/quickstart.rst]: Should we actually move installation to its own section? Could be saved for a subsequent PR.

File: docs/source/configuration.rst (hunk 6 above)
Comment (quasiben) [reply docs/source/configuration.rst]: There's an explanation of UCX_MEMTYPE_CACHE later in configuration docs

File: docs/source/configuration.rst (hunk 7 above)
Comment (quasiben) [reply docs/source/configuration.rst]: yeah

File: docs/source/configuration.rst (hunk 8 above)
Comment (quasiben) [reply docs/source/configuration.rst]: What is Unified Virtual Addressing ?

File: docs/source/configuration.rst (hunk 10 above)
Comment (quasiben) [reply docs/source/configuration.rst]: Good idea

File: docs/source/quickstart.rst (hunk 11 above)
Comment (quasiben) [reply docs/source/quickstart.rst]: done

File: docs/source/configuration.rst (hunk 8 above)
Comment (jakirkham) [reply docs/source/configuration.rst]: Basically a single address space is used for all host and device memory. This [developer doc section]( https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#unified-virtual-address-space ) provides a good explanation.

Comment (jakirkham) [reply docs/source/configuration.rst]: That said, it sounds like based on our discussion earlier we should punt on this. I'll raise an issue after we merge.

**Pull request merged**